src/main/java/Main.java

```diff
@@ -0,0 +1,186 @@
+                    }
+                pageNumber++;
+            }
+
+
+            m.put("pageNumber", pageNumber);
+            m.put("userRestaurants", userRestaurants);
+            m.put("currentPageList", currentPageList);
+            m.put("user", userName);
+
+            return new ModelAndView(m, "home.html");
+        }, new MustacheTemplateEngine());
+
+        Spark.post("/create-user" , (request, response) -> {
+            String name = request.queryParams("loginName");
+            User user = users.get(name);
+
+            if( user == null && !(users.containsKey(user))){
+                user = new User( name);
+                users.put(name, user);
+            }
+
+            Session session = request.session();
+            session.attribute(SESSION_USERNAME, name);
+
+            response.redirect("/");
+            return "";
+        });
+
+        Spark.post("/logout" ,(request, response) -> {
+            request.session().invalidate();
+            response.redirect("/");
+            return "";
+        });
+
+        Spark.post("/create-restaurant", (request, response) -> {
+            Session session = request.session();
+            String userName = session.attribute(SESSION_USERNAME);
+
+            String name = request.queryParams("restaurantName");
+            String menuType = request.queryParams("menuType");
+            String address = request.queryParams("restaurantAddress");
+            if(request.queryParams("restaurantRating").isEmpty()) {
+                response.redirect("/");
+            }
+            String userRating = request.queryParams("restaurantRating");
+            int rating = Integer.valueOf(userRating);
+            if(request.queryParams().isEmpty()){
+                response.redirect("/");
+            }
+
+            id++;
+
+            Restaurant entry = new Restaurant(name, menuType, address, rating, userName, id);
+            restaurants.add(entry);
+
+            String url = "/";
+
+            if (id >= 0){
+                url += "?id="+id;
+            }
+
+            response.redirect(url);
+            return "";
+        });
+
+        Spark.get("/edit", (request, response) -> {
+            String restaurantId = request.queryParams("restaurantId");
+            int id = Integer.valueOf(restaurantId);
+
+            Map m = new HashMap();
+            for(Restaurant restaurant: restaurants){
+                if(restaurant.getRestaurantId()  == id){
+                    m.put("userRestaurant", restaurant);
+                }
+            }
+            return new ModelAndView(m, "edit.html");
+        }, new MustacheTemplateEngine());
+
+        Spark.post("/edit-restaurant", (request, response) -> {
+            Map m = new HashMap();
+            Session session = request.session();
+            String userName = session.attribute(SESSION_USERNAME);
+
+            String idString = request.queryParams("restaurantId");
```

Comment:
Largely the same code exists above.  This should be broken out into a method of its own.